### DISCUSSION TITLE
View port panel / window

### DISCUSSION THREAD

**Discussion by llfab** [StereoKit/StereoKit#297] [Q&A]
I understand how panels / windows are placed in the MR scene using the pose. I want to create a scrollable text output panel that by definition shall be in view port space (not in scene space). This panel shall be used to output colored logging text. Is there support for this in SK without creating a window and re-position the window constantly to view port?

**Comment by maluoi** [comment]
You can't _exactly_ use viewport space with SK, but you can head-lock UI content. To do that, you can use something like use a `Hierarchy.Push(Input.Head.ToMatrix())` do your UI, then `Hierarchy.Pop()`. If the UI gives you trouble for any reason with `Hierarchy.Push`, you could probably also use `UI.Push/PopSurface` the same way :)

If you just want to dump out some text, you can also just call `Text.Add` without using the UI stuff!

**Reply by llfab** [reply]
Thanks Nick, I will give that a try. So far, I love the framework because I do not need to take care about all the OpenXR low level stuff, keep doing .net and do not get into longish turnaround times like with Unity. Good job!
